src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java

```diff
@@ -87,39 +90,58 @@ public class DependenciesFinder
   public Set<ResolvedDependency> findResolvedDependencies(
       Project project,
       boolean allConfigurations,
-      Map<String, String> variantAttributes)
+      Map<String, String> variantAttributes,
+      boolean excludeCompileOnlyDependencies)
   {
+    Set<String> compileOnlyDependenciesIds =
+        excludeCompileOnlyDependencies ? getCompileOnlyDependencyIds(project) : Collections.emptySet();
+
     return new LinkedHashSet<>(new LinkedHashSet<>(project.getConfigurations()).stream()
         .filter(configuration -> isAcceptableConfiguration(configuration, allConfigurations))
-        .flatMap(configuration -> getDependencies(project, configuration, variantAttributes,
-            resolvedConfiguration -> resolvedConfiguration.getFirstLevelModuleDependencies().stream()))
+        .flatMap(configuration -> {
+          Stream<ResolvedDependency> dependencies = getDependencies(project, configuration, variantAttributes,
+              resolvedConfiguration -> resolvedConfiguration.getFirstLevelModuleDependencies().stream());
```

Comment:
Maybe extract to a constant? 
```
 public static final Function<ResolvedConfiguration, Stream<ResolvedDependency>> RESOLVED_CONFIGURATION_STREAM_FUNCTION =
      resolvedConfiguration -> resolvedConfiguration.getFirstLevelModuleDependencies().stream();
```

Reply:
> public static final Function<ResolvedConfiguration, Stream<ResolvedDependency>> RESOLVED_CONFIGURATION_STREAM_FUNCTION =
>       resolvedConfiguration -> resolvedConfiguration.getFirstLevelModuleDependencies().stream();

Done: 2255a1205c768213b96e84ccaf5c0e7974e516bc